contrib/cirrus/setup_environment.sh

```diff
@@ -236,7 +236,7 @@ case "$TEST_FLAVOR" in
         # Use existing host bits when testing is to happen inside a container
         # since this script will run again in that environment.
         # shellcheck disable=SC2154
-        if ((CONTAINER==0)) && [[ "$TEST_ENVIRON" == "host" ]]; then
+        if ((CONTAINER==0)); then
```

Comment:
@cevich @edsantiago PTAL. This was required to build the new binary on the host and then I mount it in the container for the tests. I think this change is correct, otherwise this was never called for container tests since `TEST_ENVIRON` was set to `container`

Reply:
Right, this is because `setup_environment.sh` is called by `runner.sh` (a second time) when testing is to happen inside a container:

https://github.com/containers/podman/pull/11565/files#diff-db575befc56ec85f83b824430b9b62e20eb19d08b351dc743db26848639896a6R126

So the volume mount for `/var/libexec/podman` doesn't seem right either, normally the podman binary for testing would be compiled + installed inside the container environment (again, by way of calling `setup_environment.sh` again).

Note: I haven't seen the rest of this PR, so I'm not sure what it's all about.  I'll take a look now...